docs/hugo/content/design/ADR-2025-01-ARM-References/_index.md

```diff
@@ -0,0 +1,230 @@
+* PRO: Solves all three scenarios.
+* CON: Problematic to have _two_ properties that handle _three_ use-cases (e.g. use `privateLinkServiceReference` for GVK or ARMId, and `privateLinkServiceId` for alias).
+* CON: No control over the naming of things
+
+### Option 3: Custom reference types
+
+Instead of always replacing the property with a `genruntime.ResourceReference`, allow selection between a set of standard reference types.
+
+For the vast majority of reference properties, we'd continue to use the existing `genruntime.ResourceReference`.
+
+To support the _Alias_ and _Friendly names_ cases, introduce a new reference type, say `genruntime.AliasResourceReference`, with the following structure:
+
+``` go
+type AliasResourceReference struct {
+   Group     string
+   Kind      string
+   Namespace string
+   Name      string
+
+    ARMID string
+
+    Alias string
+}
+```
+
+The code generator would need to be updated to support this new value.
+
+Finally, when the property isn't an reference at all, but just a simple string, we'd have a value that means _leave this alone_. This is the case where we'd currently us `$armReference: false`.
+
+Again, our current `$armReference` flag would need to be renamed to something more general. Perhaps `$referenceType` with possible values `arm`, `armOrAlias`, and `other`.
+
+* PRO: Great YAML structure for end users of resources.
+* PRO: Extensible for the (admittedly unlikely) scenario where we need another different kind.
+* PRO: Cleanly handles both the _Alias_ and _Friendly names_ scenarios.
+* CON: Slightly greater implementation complexity.
+* CON: Does not solve the _Missed ARM References_ problem.
+
+### Option 4: Multivalued custom reference types
+
+Similar to Option 3, but allowing set combinations of values to be specified for the reference type.
+
+Possible values would be `arm`, `alias` and `other`, plus the combinations `arm+alias` and `arm+other`. Other combinations would be invalid.
+
+Returning to the `PrometheusRuleGroupAction` example from above, we'd specify `$referenceType: arm+other` to get:
+
+``` go
+type PrometheusRuleGroupAction struct {
+    ActionGroupId        string
+    ActionGroupReference *genruntime.ResourceReference
+    ActionProperties     map[string]string
+}
+```
+
+This handles the _Missed ARM References_ problem by allowing us to retain the existing property while introducing the corrected property side-by-side.
+
+In the `RoleAssignment` case, we'd specify `$referenceType: arm+alias` to get:
+
+``` go
+type RoleAssignment_Spec struct {
+    // ... elided ...
+    roleDefinitionReference *genruntime.AliasResourceReference
+}
+```
+
+* PRO: Great YAML structure for end users.
+* PRO: Good experience for configuration of our code generator.
+* PRO: Extensible for the (admittedly unlikely) scenario where we need another different kind.
+* PRO: Solves all three scenarios.
+* CON: Slightly greater implementation complexity.
```

Comment:
As discussed: We'll want to look into where we resolve IDs and make sure that implementing this as a per-resource extension (versus someplace else, though not sure where...) is right.